source/MaterialXTest/ShaderValid.cpp

```diff
@@ -278,16 +300,28 @@ TEST_CASE("GLSL Validation from HwShader", "[shadervalid]")
 
     // Setup lighting
     mx::LightHandlerPtr lightHandler = mx::LightHandler::creator();
-    createLightRig(doc, *lightHandler, static_cast<mx::HwShaderGenerator&>(*shaderGenerator));
+    mx::HwShaderGenerator& hwGenerator = static_cast<mx::HwShaderGenerator&>(*shaderGenerator);
+    createLightRig(doc, *lightHandler, hwGenerator);
+    // Pre-clamp the number of light sources to the number bound
+    size_t lightSourceCount = lightHandler->getLightSources().size();
```

Comment:
Niklas, I reverted the default max light count back to 3 (as before). I believe your strategy is the most flexible. Allow setting to a max so that a shader could be reused but we only bind some lights. This will use up more uniform slots in a hw shader but it's an explicit app decision. For the test I have added in this extra setting which does the clamp to the number of actual lights bound for testing. When we look at app integration we can decide what to do: 1) either recompile when the light count changes or 2) just change the number of lights bound and reuse the shader.